docs/config-validation.md

```diff
@@ -0,0 +1,191 @@
+# Configuration Validation
+
+We use a fork of
+[github.com/go-playground/validator](https://github.com/go-playground/validator)
+to enable validation of our configuration files using struct tags. The validator
+is configured to validate the configuration file at component startup if the
+`-validate` command line flag is set. If the supplied configuration file is
+valid, the component will exit with a zero exit code, otherwise it will exit
+with a non-zero exit code and print the validation errors to stderr.
+
+## Struct Tag Tips
+
+You can find the full list of struct tags supported by the validator [here]
+(https://pkg.go.dev/github.com/go-playground/validator/v10#section-documentation).
```

Comment:
Either here or in the previous paragraph, probably worth linking to our actual fork, instead of to upstream both times.